packages/features/settings/BookerLayoutSelector.tsx

```diff
@@ -0,0 +1,192 @@
+import * as RadioGroup from "@radix-ui/react-radio-group";
+import { Trans } from "next-i18next";
+import Link from "next/link";
+import { useCallback, useState } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { BookerLayouts } from "@calcom/prisma/zod-utils";
+import { bookerLayoutOptions, type BookerLayoutSettings } from "@calcom/prisma/zod-utils";
+import useMeQuery from "@calcom/trpc/react/hooks/useMeQuery";
+import { Label, Checkbox, Button } from "@calcom/ui";
+
+type BookerLayoutSelectorProps = {
+  title?: string;
+  description?: string;
+  name?: string;
+  /**
+   * If this boolean is set, it will show the user settings if the event does not have any settings (is null).
+   * In that case it also will NOT register itself in the form, so that way when submitting the form, the
+   * values won't be overridden. Because as long as the event's value is null, it will fallback to the user's
+   * settings.
+   */
+  fallbackToUserSettings?: boolean;
+};
+
+export const BookerLayoutSelector = ({
+  title,
+  description,
+  name,
+  fallbackToUserSettings,
+}: BookerLayoutSelectorProps) => {
+  const { control, getValues } = useFormContext();
+  const { t } = useLocale();
+  // Only fallback if event current does not have any settings, and the fallbackToUserSettings boolean is set.
+  const shouldShowUserSettings =
+    (fallbackToUserSettings && getValues(name || "bookerLayouts") === null) || false;
+
+  // @TODO: Enable this before merge, so we don't show the settings before release.
+  // const flags = useFlagMap();
```

Comment:
Added a feature toggle here. That way we can decide if we want to disable / enable the layout settings via a toggle :) Disabled it for now, because the migrations haven't run on the preview env yet. 

- [ ]  So todo is to enable this flag before merge.